forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -2939,6 +2939,12 @@ public static final List<SpellAbility> getBasicSpellsFromPlayEffect(final Card t
                 la.setCardState(original);
                 list.add(la);
             }
+            if (state == CardStateName.Transformed && tgtCard.isPermanent() && !tgtCard.isAura()) {
+                // casting defeated battle
+                Spell sp = new SpellPermanent(tgtCard);
```

Comment:
Use this constructor:

https://github.com/Card-Forge/forge/blob/master/forge-game/src/main/java/forge/game/spellability/SpellPermanent.java#L53